wolfcrypt/src/integer.c

```diff
@@ -2478,6 +2478,7 @@ int fast_mp_montgomery_reduce (mp_int * x, mp_int * n, mp_digit rho)
   if (W == NULL)
     return MP_MEM;
 #endif
+  XMEMSET(W, 0, sizeof(mp_word) * MP_WARRAY);
```

Comment:
Large impact on ECC performance.

Putting in a check that MP_WARRAY >= larger n->used * 2 + 1 at start will also fix scan-build.

Reply:
I fought with this a bit. Scan-build really thinks there's a problem here. Zeroing out the chunk of W we will be using is the solution to this problem that seems to work.